aptos-move/framework/aptos-experimental/sources/trading/order_book/bulk_order_book_types.move

```diff
@@ -0,0 +1,617 @@
+            total_remaining_ask_size: u64, // remaining size of the ask order
+            bid_prices: vector<u64>, // prices for each levels of the order
+            bid_sizes: vector<u64>, // sizes for each levels of the order
+            ask_prices: vector<u64>, // prices for each levels of the order
+            ask_sizes: vector<u64>, // sizes for each levels of the order
+        }
+    }
+
+    /// Represents the result of an order matching operation.
+    ///
+    /// Contains details about a matched order including the price, original size,
+    /// remaining size, and whether it was a bid or ask order.
+    ///
+    /// # Fields:
+    /// - `order_id`: Unique identifier for the order
+    /// - `account`: Account that placed the order
+    /// - `unique_priority_idx`: Priority index for time-based ordering
+    /// - `price`: Price at which the order was matched
+    /// - `orig_size`: Original size of the order
+    /// - `remaining_size`: Remaining size after the match
+    /// - `is_bid`: True if this was a bid order, false if ask order
+    enum BulkOrderResult has copy, drop {
+        V1 {
+            order_id: OrderIdType,
+            account: address,
+            unique_priority_idx: UniqueIdxType,
+            price: u64,
+            orig_size: u64,
+            remaining_size: u64,
+            is_bid: bool,
+        }
+    }
+
+    /// Represents a single match between a taker order and a maker order.
+    ///
+    /// Contains the matched order details and the size that was matched in this
+    /// particular match operation.
+    ///
+    /// # Fields:
+    /// - `order`: The matched order result
+    /// - `matched_size`: The size that was matched in this operation
+    enum SingleBulkOrderMatch has drop, copy {
+        V1 {
+            order: BulkOrderResult,
+            matched_size: u64
+        }
+    }
+
+    /// Creates a new bulk order with the specified parameters.
+    ///
+    /// # Arguments:
+    /// - `order_id`: Unique identifier for the order
+    /// - `account`: Account placing the order
+    /// - `unique_priority_idx`: Priority index for time-based ordering
+    /// - `bid_prices`: Vector of bid prices in descending order
+    /// - `bid_sizes`: Vector of bid sizes corresponding to each price level
+    /// - `ask_prices`: Vector of ask prices in ascending order
+    /// - `ask_sizes`: Vector of ask sizes corresponding to each price level
+    ///
+    /// # Returns:
+    /// A new `BulkOrder` instance with calculated original and remaining sizes.
+    public(friend) fun new_bulk_order(
+        order_id: OrderIdType,
+        account: address,
+        unique_priority_idx: UniqueIdxType,
+        bid_prices: vector<u64>,
+        bid_sizes: vector<u64>,
+        ask_prices: vector<u64>,
+        ask_sizes: vector<u64>
+    ): BulkOrder {
+        // Original bid and ask sizes are the sum of the sizes at each price level
+        let orig_bid_size = bid_sizes.fold(0, |acc, size| acc + size);
```

Comment:
Do you need to validate the ask/px are in monotonic increase order? I saw the logic is implemented in `validate_mm_order_request` but was wondering if that should apply to all bulk orders.

Reply:
Yes, good point. I moved all validation to a single place in the constructor now.